ibis/expr/operations/udf.py

```diff
@@ -149,15 +222,73 @@ def pyarrow(fn: Callable | None = None, *args: Any, **kwargs: Any) -> Callable:
         - [`python`](./scalar-udfs.qmd#ibis.expr.operations.scalar.python)
         - [`pandas`](./scalar-udfs.qmd#ibis.expr.operations.scalar.pandas)
         """
-        return _wrap(scalar._make_wrapper, InputType.PYARROW, fn, *args, **kwargs)
+        return _wrap(
+            scalar._make_wrapper,
+            InputType.PYARROW,
+            fn,
+            *args,
+            target=target,
+            schema=schema,
+            **kwargs,
+        )
 
+    @util.experimental
     @staticmethod
-    def _opaque(fn: Callable | None = None, *args: Any, **kwargs: Any) -> Callable:
-        """Construct a scalar user-defined function that is defined outside of Python."""
-        return _wrap(scalar._make_wrapper, InputType.OPAQUE, fn, *args, **kwargs)
+    def builtin(
+        fn: Callable | None = None,
+        *args: Any,
+        target: str | None = None,
+        schema: str | None = None,
+        **kwargs: Any,
+    ) -> Callable:
+        """Construct a scalar user-defined function that is built-in to the backend.
+
+        Parameters
+        ----------
+        fn
+            The The function to wrap.
+        args
+            Configuration arguments for the UDF.
+        target
+            The name of the UDF in the backend if different from the function name.
```

Comment:
Thoughts on renaming this to `name`? `target` doesn't feel intuitive to me.

Reply:
I used `target` because I am trying to surface the idea that this value will be the name the function compiles to. `name` seems more intuitive, but IMO too vague to convey what I am going for.

Reply:
How about `compiles_to`?

Reply:
Ok, renamed to `compiles_to`.

Reply:
Ok, I ended up going with your suggestion @jcrist. It's a lot less vague-looking than I thought it would be1